src/commands/use.ts

```diff
@@ -62,48 +69,36 @@ export const command = new Command("use [alias_or_project_id]")
       const hasAlias = options.rc.hasProjectAlias(newActive);
       const resolvedProject = options.rc.resolveAlias(newActive);
       validateProjectId(resolvedProject);
-      return getFirebaseProject(resolvedProject)
-        .then((foundProject) => {
-          project = foundProject;
-        })
-        .catch(() => {
-          return utils.reject("Invalid project selection, " + verifyMessage(newActive));
-        })
-        .then(() => {
-          if (aliasOpt) {
-            // firebase use [project] --alias [alias]
-            if (!project) {
-              return utils.reject(
-                "Cannot create alias " + clc.bold(aliasOpt) + ", " + verifyMessage(newActive),
-              );
-            }
-            options.rc.addProjectAlias(aliasOpt, newActive);
-            logger.info("Created alias", clc.bold(aliasOpt), "for", resolvedProject + ".");
-          }
+      if (aliasOpt) {
+        // firebase use [project] --alias [alias]
+        if (!project) {
+          throw new FirebaseError(
+            "Cannot create alias " + clc.bold(aliasOpt) + ", " + verifyMessage(newActive),
+          );
+        }
+        options.rc.addProjectAlias(aliasOpt, newActive);
+        logger.info("Created alias", clc.bold(aliasOpt), "for", resolvedProject + ".");
+      }
 
-          if (hasAlias) {
-            // found alias
-            if (!project) {
-              // found alias, but not in project list
-              return utils.reject(
-                "Unable to use alias " +
-                  clc.bold(newActive) +
-                  ", " +
-                  verifyMessage(resolvedProject),
-              );
-            }
+      if (hasAlias) {
+        // found alias
+        if (!project) {
+          // found alias, but not in project list
+          throw new FirebaseError(
+            "Unable to use alias " + clc.bold(newActive) + ", " + verifyMessage(resolvedProject),
+          );
+        }
 
-            utils.makeActiveProject(options.projectRoot, newActive);
-            logger.info("Now using alias", clc.bold(newActive), "(" + resolvedProject + ")");
-          } else if (project) {
-            // exact project id specified
-            utils.makeActiveProject(options.projectRoot, newActive);
-            logger.info("Now using project", clc.bold(newActive));
-          } else {
-            // no alias or project recognized
-            return utils.reject("Invalid project selection, " + verifyMessage(newActive));
-          }
-        });
+        makeActiveProject(options.projectRoot, newActive);
+        logger.info("Now using alias", clc.bold(newActive), "(" + resolvedProject + ")");
+      } else if (project) {
```

Comment:
I think the tests are failing since `project` never gets assigned a value. It looks like we assigned the value using https://github.com/firebase/firebase-tools/blob/495cb72c8e7a4270a4e9d602fddc9ee8fdca53d6/src/commands/use.ts#L65-L68, but we removed it here since `getFirebaseProject` requires the user to be logged in .

It might be possible to change `project` -> `resolvedProject`